src/ui/pages/SwapForm/SlippageSettings/SlippageSettings.tsx

```diff
@@ -91,6 +95,67 @@ function SlippageWarning({ percentValue }: { percentValue: string }) {
   );
 }
 
+function PercentWidth({ onValue }: { onValue(width: number): void }) {
+  const persentCharWidthRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    let interval: ReturnType<typeof setTimeout>;
+    if (persentCharWidthRef.current) {
+      onValue(persentCharWidthRef.current.clientWidth);
+    } else {
+      interval = setTimeout(
+        () => onValue(persentCharWidthRef.current?.clientWidth || 0),
+        100
+      );
+    }
+    return () => clearTimeout(interval);
+  }, [onValue]);
+
+  return (
+    <UIText
+      ref={persentCharWidthRef}
+      kind="body/accent"
```

Comment:
maybe extract this `kind` value to a variable so that it's explicitly passed to be the same here and in `CustomValueOverlay`?